crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -2043,6 +2044,89 @@ impl OlmMachine {
         DehydratedDevices { inner: self.to_owned() }
     }
 
+    /// Get the stored encryption settings for the given room, such as the
+    /// encryption algorithm or whether to encrypt only for trusted devices.
+    ///
+    /// These settings can be modified via [`set_room_settings`].
+    pub async fn get_room_settings(&self, room_id: &RoomId) -> StoreResult<Option<RoomSettings>> {
+        // There's not much to do here: it's just exposed for symmetry with
+        // `set_room_settings`.
+        self.inner.store.get_room_settings(room_id).await
+    }
+
+    /// Store encryption settings for the given room.
+    ///
+    /// This method checks if the new settings are "safe" -- ie, that they do
+    /// not represent a downgrade in encryption security from any previous
+    /// settings. Attempts to downgrade security will result in a
+    /// [`SetRoomSettingsError::EncryptionDowngrade`].
+    ///
+    /// If the settings are valid, they will be persisted to the crypto store.
+    /// These settings are not used directly by this library, but the saved
+    /// settings can be retrieved via [`get_room_settings`].
+    pub async fn set_room_settings(
```

Comment:
I'm not completely sure if this is the best place for this API: `OlmMachine` already has a million methods and adding more isn't my favourite thing. But the only other likely-looking place seems to be the Store itself, and given this is application-level logic, that doesn't feel right.

Reply:
I agree with your reasoning. Let's keep it here for the moment. We can refactor later.